tests/test_core.py

```diff
@@ -170,6 +170,132 @@ def test_create_interpolator(simple_grid):
     assert np.allclose(interp(field), linear_value)
 
 
+def test_set_cell_volumes():
+    """
+    Test that cell volumes are set properly.
+    """
+# Test cartesian volumes
+    grid_conf2 = {"r_min": 0,
+                  "r_max": 1,
+                  "dr": 0.1,
+                  "coordinate_system": "cartesian"}
+    grid2 = Grid(grid_conf2)
+    grid = grid2.cell_edges
+    volumes = grid[1:] - grid[0:-1]
+    assert grid2.cell_volumes.size == grid.size-1
+    assert np.allclose(grid2.cell_volumes, volumes)
+# Test cylindrical volumes
+    grid_conf2 = {"r_min": 0,
+                  "r_max": 1,
+                  "dr": 0.1,
+                  "coordinate_system": "cylindrical"}
+    grid2 = Grid(grid_conf2)
+    grid = grid2.cell_edges
+    volumes = np.pi*(grid[1:]**2 - grid[0:-1]**2)
+    assert grid2.cell_volumes.size == grid.size-1
+    assert np.allclose(grid2.cell_volumes, volumes)
+# Test spherical volumes
+    grid_conf2 = {"r_min": 0,
+                  "r_max": 1,
+                  "dr": 0.1,
+                  "coordinate_system": "spherical"}
+    grid2 = Grid(grid_conf2)
+    grid = grid2.cell_edges
+    volumes = 4./3.*np.pi*(grid[1:]**3 - grid[0:-1]**3)
+    assert grid2.cell_volumes.size == grid.size-1
+    assert np.allclose(grid2.cell_volumes, volumes)
+
+
+def test_set_interface_volumes():
+    """
+    Test that cell volumes are set properly.
```

Comment:
might be wrong but, should this docstring be more about interface volumes and not cell volumes? just going off of method names.